myuw/dao/enrollment.py

```diff
@@ -48,6 +71,77 @@ def get_main_campus(request):
     return campuses
 
 
+def _get_degrees_for_terms(terms, enrollments, accessor):
+    """
+    Takes in a list of terms and a dictionary of terms to enrollments (returned
+    by get_all_enrollments), and returns a list of either majors or minors
+    and their terms, depending upon which accessor is used, 'majors' or
+    'minors'
+    """
+    degrees = []
+    previous = None
+
+    for term in terms:
+        if (term in enrollments and
+                len(getattr(enrollments[term], accessor)) > 0):
+            enrollment = enrollments[term]
+            entry = {}
+            entry['quarter'] = term.quarter
+            entry['year'] = term.year
+
+            term_degrees = getattr(enrollments[term], accessor)
+            entry[accessor] = []
+
+            if previous is not None:
```

Comment:
can you change the initial state of previous so you don't have a special case here?